src/bin/cmd/client.rs

```diff
@@ -117,7 +127,16 @@ pub fn unban_peer(config: &ServerConfig, peer_addr: &SocketAddr) {
 pub fn list_connected_peers(config: &ServerConfig) {
 	let mut e = term::stdout().unwrap();
 	let url = format!("http://{}/v1/peers/connected", config.api_http_addr);
-	match api::client::get::<Vec<p2p::PeerInfo>>(url.as_str()).map_err(|e| Error::API(e)) {
+	let peers_info: Result<Vec<p2p::PeerInfo>, api::Error>;
+	if config.api_basic_auth.is_some() && config.api_basic_auth.unwrap() {
```

Comment:
What if we hide this logic in the client API?
* add param `api_secret: Option<String>` to `api::client::get/post
* generate `api_basic_auth` in the client's request builder
* Set `AUTHORIZATION` header if `api_secret` param is not `None`

We may be able to remove all auth-specific methods (`build/get/post*_basic_auth`) in the client and don't need any specific handling here except blindly passing `config.api_basic_auth`

Reply:
I did what you suggested. Feel free to take another look :).